### PR TITLE
`CheckoutThankYouHeader`: refactor tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by flootr**
#### Proposed Changes

* `CheckoutThankYouHeader`: refactor tests to `@testing-library/react`

#### Testing Instructions

* Verify unit tests still pass

Related to #63409

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/test/header.js [hunk 1]
```diff
@@ -26,50 +27,57 @@ describe( 'CheckoutThankYouHeader', () => {
...
 		test( 'Should display getText()-based success message when isSimplified=false (default)', () => {
-			const comp = shallow( <CheckoutThankYouHeader isDataLoaded={ true } { ...defaultProps } /> );
-			expect( comp.find( '.checkout-thank-you__header-heading' ).text() ).toEqual(
+			render( <CheckoutThankYouHeader isDataLoaded={ true } { ...defaultProps } /> );
+			expect( screen.getByRole( 'heading', { level: 1 } ) ).toHaveTextContent(
 				'Congratulations on your purchase!'
 			);
-			expect( comp.find( '.checkout-thank-you__header-text' ).text() ).toEqual(
-				"Your site is now on the {{strong}}%(productName)s{{/strong}} plan. It's doing somersaults in excitement!"
+			expect( screen.getByRole( 'heading', { level: 2 } ) ).toHaveTextContent(
```
Comment (sirbrillig) [inline client/my-sites/checkout/checkout-thank-you/test/header.js:45]: TIL you can look for headings by level. Very cool.

**Review by sirbrillig — APPROVED** [review]
👍 from a Payments team perspective.

File: client/my-sites/checkout/checkout-thank-you/test/header.js [hunk 2]
```diff
@@ -26,50 +27,57 @@ describe( 'CheckoutThankYouHeader', () => {
...
 			);
-			expect( comp.find( '.checkout-thank-you__header-text' ).length ).toBe( 0 );
-			expect( comp.find( '.checkout-thank-you__success-message-item' ).length ).toEqual( 2 );
-			expect(
-				comp.find( '.checkout-thank-you__success-message-item:last-child div' ).text()
-			).toEqual(
-				"Your site has been launched. You can share it with the world whenever you're ready."
+			expect( screen.queryByRole( 'heading', { level: 2 } ) ).not.toBeInTheDocument();
+
+			const messages = screen.queryAllByRole( 'listitem' );
```
Comment (tyxla) [inline client/my-sites/checkout/checkout-thank-you/test/header.js:74]: Could we also use an assertion to ensure the number of list items found by the query here is correct?

**Review by tyxla — APPROVED** [review]
Left one small remark, but other than that, this looks great 👍 🚀

File: client/my-sites/checkout/checkout-thank-you/test/header.js (hunk 2 above)
Comment (flootr) [reply client/my-sites/checkout/checkout-thank-you/test/header.js:74]: Done via https://github.com/Automattic/wp-calypso/pull/66108/commits/389a1c78df9e67d2f8e8eba2e9d86fc1704b462d

**Pull request merged**